src/EventListener/PrettyErrorScreenListener.php

```diff
@@ -68,12 +74,18 @@ class PrettyErrorScreenListener
      * @param bool              $prettyErrorScreens True to render the error screens
      * @param \Twig_Environment $twig               The twig environment
      * @param ConfigAdapter     $config             The config adapter
+     * @param LoggerInterface   $logger             The logger service
      */
-    public function __construct($prettyErrorScreens, \Twig_Environment $twig, ConfigAdapter $config)
-    {
+    public function __construct(
+        $prettyErrorScreens,
+        \Twig_Environment $twig,
+        ConfigAdapter $config,
+        LoggerInterface $logger
```

Comment:
As far as I could see in the Symfony code, the Logger is always an optional dependency (https://github.com/symfony/symfony/search?utf8=✓&q=%22%24logger+%3D+null%22&type=Code). This would also prevent a (arguably irrelevant) BC break.
I guess the logger will be NULL if logging should be disabled (we did have that option in Contao 3.5).